src/Backups/BackupsWorker.cpp

```diff
@@ -18,37 +18,86 @@
 #include <Common/Exception.h>
 #include <Common/Macros.h>
 #include <Common/logger_useful.h>
-#include <Common/scope_guard_safe.h>
 #include <Common/setThreadName.h>
 
 
 namespace DB
 {
 
-namespace ErrorCodes
-{
-    extern const int LOGICAL_ERROR;
-}
-
 namespace
 {
     /// Coordination status meaning that a host finished its work.
-    constexpr const char * kCompletedCoordinationStatus = "completed";
+    constexpr const char * kCompletedStage = "completed";
+
+    std::shared_ptr<IBackupCoordination> makeBackupCoordination(const String & coordination_zk_path, const ContextPtr & context, bool is_internal_backup)
+    {
+        if (!coordination_zk_path.empty())
```

Comment:
Both variants don't look good actually:

```
if (!coordination_zk_path.empty())
{
    // do something with coordination_zk_path
    ... 
}
else
{
}
```

and

```
if (coordination_zk_path.empty())
{
    ... 
}
else
{
    // do something with coordination_zk_path
    ... 
}
```
(because in the latter case the condition about `coordination_zk_path` and actions about `coordination_zk_path` are written separately from each other)